tensorboard/plugins/scalar/tf_scalar_dashboard/tf-scalar-dashboard.html

```diff
@@ -277,8 +288,43 @@ <h3>No scalar data was found.</h3>
         return _smoothingWeight > 0;
       },
 
-      _shouldOpen(index) {
-        return index <= 2;
+      _getCategoryKey(category) {
+        return category.metadata.type ==
+            tf_categorization_utils.CategoryType.SEARCH_RESULTS ?
+                'search' : category.name;
+      },
+
+      _updatePaneOpened() {
+        const paneOpen = this._paneOpened;
+        const nextPaneOpen = {};
+
+        // Copy over the relevant values from the old pane state.
```

Comment:
FWIW, is it actually necessary to make a copy, or could we just mutate the pane state directly?  Nothing is observing this._paneOpened so I think we could just iterate over it and delete any key that isn't in the set of current category keys.  Optional but maybe more direct.

Also, we could get the removed categories directly from Polymer if we wanted to avoid recomputing the changes to the array ourselves: https://www.polymer-project.org/1.0/docs/devguide/observers#array-observation

Reply:
> FWIW, is it actually necessary to make a copy, or could we just mutate the pane state directly? Nothing is observing this._paneOpened so I think we could just iterate over it and delete any key that isn't in the set of current category keys. Optional but maybe more direct.

That's one way but AFAICT, we are taking a bit more functional approach in state manipulation (minus the recent array manipulation I introduced for performance).